compiler/circle-mpqsolver/src/core/TestHelper.test.cpp

```diff
@@ -138,5 +138,32 @@ bool isFileExists(const std::string &path)
 }
 
 } // namespace io_utils
+
+namespace hdf5_utils
+{
+
+void createHDF5File(char *name_template, uint32_t channels, uint32_t width, uint32_t height)
+{
+  io_utils::makeTemporaryFile(name_template);
+
+  const auto rank = 4;
+  hsize_t dim[4] = {1, height, width, channels};
+  H5::DataSpace space(rank, dim);
+
+  // let it be just zero
+  std::vector<float> data(height * width * channels, 0.f);
+
+  // Create test file in the current directory
+  H5::H5File file(name_template, H5F_ACC_TRUNC);
+  {
+    file.createGroup("/value");
+    file.createGroup("/value/0");
+    H5::DataSet dataset(file.createDataSet("/value/0/0", H5::PredType::IEEE_F32BE, space));
+    dataset.write(data.data(), H5::PredType::IEEE_F32LE);
```

Comment:
Q) dataset is created with `IEEE_F32BE`, but written with `IEEE_F32LE`.
is this OK?

Reply:
Ahh. No. Sorry.

Reply:
Actually seems like correct version will be `dataset.write(data.data(), H5::PredType::NATIVE_FLOAT);` because second parameter is layout of `data` to write.